mypy/semanal.py

```diff
@@ -160,6 +160,45 @@
 Tag = int
 
 
+def substitute_unbounds(t: UnboundType, names_to_types: Mapping[str, Instance]) -> ProperType:
+    """Replaces all type vars types with whatever ProperTypes given.
+
+    Used with --inherit-signatures option
+    Example, if we have:
+        class A(Generic[T,U]):
+            def foo(self) -> U:pass
+            def bar(self,*args)->Callable[[U],T]:pass
+        class B(A[int,str]):
+            def foo(self):pass
+            def bar(self,*args):pass
+
+    Then class B will be transformed to:
+         class B(A[int,str]):
+            def foo(self) -> str:pass
+            def bar(self,*args)->Callable[[str],int]:pass
```

Comment:
Please use PEP 8 everywhere including in docstring and in tests! (You can however skip empty lines in tests whenever it makes sense).